theano/compile/debugmode.py

```diff
@@ -945,7 +952,10 @@ def _find_bad_optimizations0(order, reasons, r_vals):
                 r_val = r_vals[r]
                 assert r.type == new_r.type
 
-                if hasattr(new_r, 'values_eq_approx'):
+                if hasattr(new_r.tag, 'values_eq_approx'):
```

Comment:
Because cloning doesn't keep random attributes you graft on variables.  But it does preserve the contents of tag.